### ISSUE TITLE
WebApiToSwaggerGenerator not respecting mimetypes per operation

### ISSUE THREAD

**Issue by MattFenner** [RicoSuter/NSwag#2147]
Hi,

We are using the WebApiToSwaggerGenerator.GenerateForControllersAsync(controllerTypes) method to generate our swagger file, but it doesn't seem to respect any ProducesAttribute put on any of the rest endpoint methods.

For example we have a method defined:
`[HttpGet("ds/csv")]
        [SwaggerResponse(HttpStatusCode.OK, typeof(CsvSwaggerFeatureResponse<DSRealtimeCsvEntry>))]
        [SwaggerResponse(HttpStatusCode.Forbidden, typeof(string))]
        [SwaggerTag("Traffic Data Service")]
        [Description("Get Detector Site Real-time Measures")]
        [LongDescription("Returns the realtime traffic measures for detector sites.")]
        [Produces("text/csv")]
        public async Task<IActionResult> GetDsRealtime()`

I would expect the generated swagger should be something like:
`
"paths": {
    "/traffic/v1/ds/csv": {
      "get": {
        "tags": [
          "Traffic Data Service"
        ],
        "summary": "Get Detector Site Real-time Measures",
        "produces": [
            "text/csv"
        ],
        "operationId": "Traffic_GetDsRealtime",`

but it is missing the produces section on the operation.

From what I can tell this feature is implemented in some of the other Generator classes. Would it be possible to also implement it on WebApiToSwaggerGenerator. Or are we forced to 'upgrade' to one of the other generators?

Thanks

**Comment by darman96** [comment]
noticed the same behavior, the only mime types which worked for me were "application/json", "text/json" and "text/plain".

**Comment by RicoSuter** [comment]
WebApiToSwaggerGenerator and AspNetCoreToSwaggerGenerator are different generators - i suspect that WebApiToSwaggerGenerator might not correctly support this - because it's deprecated I don't have the time to look into this... 

Do we also have the problem with the AspNetCoreToSwaggerGenerator?

**Comment by darman96** [comment]
Well how do I specify the generator?
I thought since I'm on a AspNetCore Project nswag defaults to the AspNetCoreToSwaggerGenerator.

**Comment by RicoSuter** [comment]
@darman96 how did you integrate NSwag in your Startup.cs?

**Comment by darman96** [comment]
in ConfigureContainer() i have this: 
`services.AddSwaggerDocument(
                settings =>
                {
                    settings.DocumentName = "swagger";
                });`

and in my Configure(): 
`app.UseSwagger(
                settings =>
                {
                    settings.DocumentName = "swagger";
                    settings.Path = "/swagger/v1/swagger.json";
                });

app.UseSwaggerUi3(
                settings =>
                {
                    settings.Path = "/swagger";
                    settings.DocumentPath = "/swagger/v1/swagger.json";
                });`

**Comment by RicoSuter** [comment]
Yes, this is using the AspNetCoreToSwaggerGenerator... otherwise you'd see obsolete warnings... please create another issue if its AspNetCoreToSwaggerGenerator related

**Comment by darman96** [comment]
Are you sure? Because when I hover over app.UseSwaggerUi3() I get this tool tip:
![06-05-2019 17-26-28](https://user-images.githubusercontent.com/2633918/57236027-78d6fe00-7024-11e9-922d-47a586b89817.png)

**Comment by RicoSuter** [comment]
That's still a legacy leftover and will be removed as soon as the WebApiToSwaggerGenerator is completely removed from NSwag.AspNetCore (currently there for legacy users) in v13

**Comment by MattFenner** [comment]
Thanks for your quick response Rico, we will look into upgrading to a different generator.
